frontend/pages/hosts/details/cards/HostSoftwareLibrary/HostInstallerActionCell/HostInstallerActionCell.tsx

```diff
@@ -0,0 +1,236 @@
+      if (!hostMDMEnrolled) {
+        installDisabled = true;
+        installTooltip = "To install, turn on MDM for this host.";
+      }
+    }
+  }
+
+  return {
+    installDisabled,
+    installTooltip: installTooltip || undefined,
+    uninstallDisabled,
+    uninstallTooltip: uninstallTooltip || undefined,
+  };
+};
+
+export const HostInstallerActionButton = ({
+  baseClass,
+  tooltip,
+  disabled,
+  onClick,
+  icon,
+  text,
+  testId,
+}: IHostInstallerActionButtonProps) => (
+  <div className={`${baseClass}__item-action`}>
+    <TooltipWrapper
+      tipContent={tooltip}
+      underline={false}
+      showArrow
+      position="top"
+    >
+      <Button
+        variant="text-icon"
+        type="button"
+        className={`${baseClass}__item-action-button`}
+        onClick={onClick}
+        disabled={disabled}
+      >
+        <Icon name={icon} color="core-fleet-blue" size="small" />
+        <span data-testid={testId}>{text}</span>
+      </Button>
+    </TooltipWrapper>
+  </div>
+);
+
+/** HostInstallerActionCell component has different disabled states
+ * and tooltips for Host details > Library HostInstallerActionCell than
+ * Fleet Desktop > Self-service HostInstallerActionCell */
+export const HostInstallerActionCell = ({
+  software,
+  onClickInstallAction,
+  onClickUninstallAction,
+  baseClass,
+  hostScriptsEnabled,
+  hostMDMEnrolled,
+  isMyDevicePage = false,
+}: IHostInstallerActionCellProps) => {
+  const {
+    id,
+    status,
+    software_package,
+    app_store_app,
+    ui_status,
+    installed_versions,
+  } = software;
+  const {
+    installDisabled,
+    installTooltip,
+    uninstallDisabled,
+    uninstallTooltip,
+  } = getActionButtonState({
+    hostScriptsEnabled: hostScriptsEnabled || false,
+    softwareId: id,
+    status,
+    appStoreApp: app_store_app,
+    hostMDMEnrolled,
+    softwarePackage: software_package,
+    isMyDevicePage,
+  });
+
+  // buttonDisplayConfig is used to track the display text and icons of the install and uninstall button
+  const [
+    buttonDisplayConfig,
+    setButtonDisplayConfig,
+  ] = useState<IButtonDisplayConfig>({
+    install: getInstallerActionButtonConfig("install", ui_status),
+    uninstall: getInstallerActionButtonConfig("uninstall", ui_status),
+  });
+
+  useEffect(() => {
+    // We update the text/icon only when we see a change to a non-pending status
+    // Pending statuses keep the original text shown (e.g. "Retry" text on failed
+    // install shouldn't change to "Install" text because it was clicked and went
+    // pending. Once the status is no longer pending, like 'installed' the
+    // text will update to "Reinstall")
+    if (status !== "pending_install" && status !== "pending_uninstall") {
+      setButtonDisplayConfig({
+        install: getInstallerActionButtonConfig("install", ui_status),
+        uninstall: getInstallerActionButtonConfig("uninstall", ui_status),
+      });
+    }
+  }, [status, ui_status]);
```

Comment:
do you need state for buttonDisplayConfig and useEffect here?

since `status` and `ui_status` are from the `software` passed into props, if the software changes, than `buttonDisplayConfig` can be calculated without state or the extra useEffect step

```tsx
const  buttonsDisplayConfig = {
    install: getInstallerActionButtonConfig("install", ui_status),
    uninstall: getInstallerActionButtonConfig("uninstall", ui_status),
  })
```

does this work here?

Reply:
Yeah, I purposefully built the state and useEffect because we don't want text to change prematurely, example in comment:

    // We update the text/icon only when we see a change to a non-pending status
    // Pending statuses keep the original text shown **(e.g. "Retry" text on failed**
    // install shouldn't change to "Install" text because it was clicked and went
    // pending.** Once the status is no longer pending, like 'installed' the
    // text will update to "Reinstall")